File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/virtual_poly_runme.m
virtual_poly

d = virtual_poly.NDouble(3.5);
i = virtual_poly.NInt(2);

#
# the copy methods return the right polymorphic types
# 
dc = d.copy();
ic = i.copy();

if (d.get() != dc.get())
  error
endif

if (i.get() != ic.get())
  error
endif

virtual_poly.incr(ic);

if ((i.get() + 1) != ic.get())
  error
endif


dr = d.ref_this();
if (d.get() != dr.get())
  error
endif


#
# 'narrowing' also works
#
ddc = virtual_poly.NDouble_narrow(d.nnumber());
if (d.get() != ddc.get())
  error
endif

dic = virtual_poly.NInt_narrow(i.nnumber());
if (i.get() != dic.get())
  error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/virtual_poly_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 8;
BEGIN { use_ok('virtual_poly') }
require_ok('virtual_poly');

my $d = virtual_poly::NDouble->new(3.5);
my $i = virtual_poly::NInt->new(2);

#
# the copy methods return the right polymorphic types
# 
my $dc = $d->copy();
my $ic = $i->copy();

is($d->get(), $dc->get());

is($i->get(), $ic->get());

virtual_poly::incr($ic);

is($i->get() + 1, $ic->get());

my $dr = $d->ref_this();
is($d->get(), $dr->get());


#
# 'narrowing' also works
#
my $ddc = virtual_poly::NDouble::narrow($d->nnumber());
is($d->get, $ddc->get());

my $dic = virtual_poly::NInt::narrow($i->nnumber());
is($i->get(), $dic->get());
